skimage/_build.py

```diff
@@ -34,15 +34,25 @@ def cython(pyx_files, working_path=''):
         from Cython.Build import cythonize
     except ImportError:
         # If cython is not found, the build will make use of
-        # the distributed .c files if present
-        c_files = [f.replace('.pyx.in', '.c').replace('.pyx', '.c') for f in pyx_files]
-        for cfile in [os.path.join(working_path, f) for f in c_files]:
-            if not os.path.isfile(cfile):
-                raise RuntimeError('Cython >= %s is required to build scikit-image from git checkout' \
-                                   % CYTHON_VERSION)
+        # the distributed .c or .cpp files if present
+        c_files_used = []
+        for pyx_file in [os.path.join(working_path, f) for f in pyx_files]:
+            pyx_file = pyx_file.replace('.in', '')
```

Comment:
I like this PR in general, but I feel this part of the code path can be improved with something like:

```
def compiled_fn(f):
    base = f.replace('.pyx.in', '')
    for ext in ('.cpp', '.c'):
        fn = base + ext
        if os.path.exists(fn):
            return fn

    raise RuntimeError('...')

cxx_files = [compiled_fn(f) for f in pyx_files]
```

Reply:
Thanks @stefanv. I think the code is much improved thanks to your idea. Hopefully the CIs pass and this PR can get merged in!!